### PR TITLE
add cluster-reader sa and kubeconfig

### PR DESCRIPTION
**Pull request by mjudeikis**
Add dedicated SA and kubeconfig for Operations to use.

### CODE REVIEW

**Comment by mjudeikis** [comment]
cc: @openshift/sig-azure

File: pkg/addons/data/ServiceAccount/openshift-infra/cluster-reader.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: cluster-reader
```
Comment (pweil-) [inline pkg/addons/data/ServiceAccount/openshift-infra/cluster-reader.yaml]: how do you feel about being more specific here?  This is for Microsoft SRE, so maybe microsoft-sre?  I'm also wondering if we want to separate anything we do specifically for OSA into a different namespace.  Didn't we just get rid of the openshift-azure one? :rofl:  

What I'm wondering here is if we need to be wary of things in the future that make assumptions on what's in the infra namespace and remove or change things we've added.

Comment (mjudeikis) [reply pkg/addons/data/ServiceAccount/openshift-infra/cluster-reader.yaml]: > Didn't we just get rid of the openshift-azure one? rofl 

Was thinking about this too. My vote would be `openshift-azure` namespace and dont distinguish our ops and their ops. Just `azure-cluster-reader`. We can create a dedicated role and role binding with `azure` prefix (or postfix). And so we know that anything with prefix `azure` is our made.

Comment (kargakis) [reply pkg/addons/data/ServiceAccount/openshift-infra/cluster-reader.yaml]: +1 on using openshift-azure for OSA-specific things.

File: pkg/addons/data/Namespace/openshift-azure.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  annotations:
+    openshift.io/node-selector: ""
```
Comment (kargakis) [inline pkg/addons/data/Namespace/openshift-azure.yaml]: Is this needed?

File: pkg/addons/data/ServiceAccount/openshift-infra/cluster-reader.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
```
Comment (kargakis) [inline pkg/addons/data/ServiceAccount/openshift-infra/cluster-reader.yaml]: Needs to be removed now?

File: pkg/addons/data/Namespace/openshift-azure.yaml (hunk 2 above)
Comment (mjudeikis) [reply pkg/addons/data/Namespace/openshift-azure.yaml]: This backfired us a few times already (lack of this annotation). I would be tempted to keep it as its default.

**Pull request merged**